File: scripts/templates/ex10.py
#! /usr/bin/env python
#
# NEMO's getparam

import math
from nemopy import getparam

keyval = [
    "aaa=1\n       The aaa (int) variable",
    "bbb=10.0\n    The bbb (float) variable",
    "VERSION=1\n   11-nov-2022 PJT",
    ]

usage = "showing off getparam in python"

if __name__ == '__main__':

    p = getparam.Param(keyval, usage)
    print('aaa:',p.get("aaa"))
    print('bbb:',p.get("bbb"))


#  alternatively we can hide the objects even more?
#        nemo.nemo_main(keyval,usage)
#        nemo.getparam("aaa")
#        nemo.getrparam("bbb")

File: scripts/templates/ex10.c
/*
 *  NEMO's getparam in C
 */

#include <nemo.h>

string defv[] = {
    "aaa=1\n             some help for aaa",
    "bbb=10.0\n          some help for bbb",
    "VERSION=1.0\n       11-Nov-2022 XYZ",
    NULL,
};

string usage="example10 NEMO getparam in C";


void nemo_main()
{
  string sa = getparam("aaa");
  string sb = getparam("bbb");
  int    ia = getiparam("aaa");
  real   ra = getrparam("bbb");
  printf("string: aaa=%s bbb=%s   int: aaa=%d   real: bbb=%g\n",
	sa,  sb, ia, ra);
}

